### DISCUSSION TITLE
get relevant articles

### DISCUSSION THREAD

**Discussion by bogoliubon** [kilian-group/phantom-wiki#27] [Ideas]
Thinking about how to implement the feature to get relevant articles only from our universe: 
input is a list of queries in order to answer the question (from our sampled queries)
output is a list of people's names. 

The way I'm thinking:
for <relation> (<name>, X) <name> is relevant;
for <attribute_name> (<name>, X) <name> is relevant; 
for  <attribute_name> (X,  <attribute_value>) all the returned X should be relevant; 
for aggregate_all () type everything returned is relevant. 

Does this make sense? 
I can work on this tonight.

**Comment by albertgong1** [comment]
What is the first one? Is that the <relation_name>(, X)?
https://github.com/ag2435/phantom-wiki/blob/632d9a29332dea288b035f22b6697ee2294d7f79/src/phantom_wiki/facts/templates.py#L56

**Reply by bogoliubon** [reply]
yes!

**Comment by albertgong1** [comment]
To summarize the possible experiments to test Yoav's question:
1. include only the relevant articles from our universe (like you described above)
  a. A further variant of this setup would be to only include the "relevant articles", but ask questions which we know cannot be answered by the "relevant articles." Do the models hallucinate or do they say “I don’t know”?
3. include the relevant articles from our universe + some irrelevant articles (like @kamilest described)
4. include the articles from a small universe + the articles from a separately generated small universe, which we know is irrelevant by construction (will just need to check that the set of names in each universe are disjoint)
